integration_tests/src/lib.rs

```diff
@@ -1432,4 +1432,79 @@ mod test {
 
         Ok(())
     }
+
+    // Test that multiple upstairs can connect to a single read-only downstairs
+    #[tokio::test(flavor = "multi_thread", worker_threads = 8)]
+    async fn integration_test_multi_read_only() -> Result<()> {
+        const BLOCK_SIZE: usize = 512;
+
+        let mut opts = three_downstairs(54040, 54041, 54042, true).unwrap();
```

Comment:
These port numbers are already in use on line 907 and can cause the test to hang.

I wish I had some better way of allocating port numbers, as I do this to myself all the time.

Also, I was trying to group the "volume" tests into one section (ending at line ~986), and having their ports start with 54xxx and the guest section of tests (which is where this is) all start with 55xxx.

Reply:
Oh, and sorry, but there is some churn in this area with all the scrubber tests I added.